python/pyspark/sql/types.py

```diff
@@ -1487,36 +1451,14 @@ class Row(tuple):
     True
     """
 
-    # Remove after Python < 3.6 dropped, see SPARK-29748
```

Comment:
cc @BryanCutler to double check. Basically, I removed the logic here to don't sort keyword arguments by default, and removed `__from_dict__` related logics too.

With Python 3.6+, it (already) broke one test case here: https://github.com/apache/spark/pull/28957/files#diff-0590ca852e0e565bc489272aee36167fL890 but I guess it's fine.

Reply:
cc @zero323 too FYI. Finally we're removing this.

Reply:
Yup, this looks good. I noticed you already fixed up the test cases that if affects, so that's great!